src/lib/draw/StreetViewMode.svelte

```diff
@@ -0,0 +1,131 @@
+<script lang="ts">
+  import { onDestroy } from "svelte";
+  import type { Mode } from "./types";
+  import type { GeoJSONSource, MapMouseEvent } from "maplibre-gl";
+  import type { Feature, Point } from "geojson";
+  import { point } from "@turf/helpers";
+  import { map } from "../../stores";
+  import {
+    emptyGeojson,
+    overwriteSource,
+    overwriteLayer,
+    drawCircle,
+  } from "../../maplibre_helpers";
+
+  const thisMode = "street-view";
+
+  const circleRadiusPixels = 10;
+
+  export let mode: Mode;
+  export let changeMode: (m: Mode) => void;
+
+  let imagery: "google" | "bing" | "mapillary" = "google";
+
+  export function start() {}
+  export function stop() {
+    cursor = null;
+  }
+
+  let cursor: Feature<Point> | null = null;
+
+  $map.on("mousemove", onMouseMove);
+  $map.on("click", onClick);
+
+  onDestroy(() => {
+    $map.off("mousemove", onMouseMove);
+    $map.off("click", onClick);
+  });
+
+  // Rendering
+  let source = "street-view";
+  overwriteSource($map, source, emptyGeojson());
+  // TODO Different icon?
+  overwriteLayer($map, {
+    id: "draw-street-view",
+    source,
+    ...drawCircle("black", circleRadiusPixels, 1.0),
+  });
+
+  $: {
+    let gj = emptyGeojson();
+    if (cursor) {
+      gj.features.push(cursor);
+    }
+    ($map.getSource(source) as GeoJSONSource).setData(gj);
+  }
+
+  function onMouseMove(e: MapMouseEvent) {
+    if (mode != thisMode) {
+      return;
+    }
+
+    cursor = cursorFeature(e.lngLat.toArray());
+  }
+
+  function onClick() {
+    if (mode != thisMode) {
+      return;
+    }
+
+    let [lon, lat] = cursor.geometry.coordinates;
+    if (imagery == "google") {
+      window.open(
+        `http://maps.google.com/maps?q=&layer=c&cbll=${lat},${lon}&cbp=11,0,0,0,0`,
+        "_blank"
+      );
+    } else if (imagery == "bing") {
+      window.open(
+        `https://www.bing.com/maps?cp=${lat}~${lon}&style=x`,
+        "_blank"
+      );
+    } else if (imagery == "mapillary") {
+      window.open(
```

Comment:
This one's a bit clunky because you have to additionally click on a specific photo, but I don't think anything faster is possible without properly integrating with their API

Reply:
I'd be inclined, as the behaviour is different to the other two, to potentially drop this until we've looked at integrating their API. Just as the UX is slightly different to what you might expect.